doc/bmp/bmp.md

```diff
@@ -0,0 +1,373 @@
+# BMP for monitoring SONiC BGP info
+## High Level Design Document
+### Rev 0.1
+
+# Table of Contents
+
+  * [Revision](#revision)
+
+  * [About this Manual](#about-this-manual)
+
+  * [Definitions/Abbreviation](#definitionsabbreviation)
+ 
+  * [1 Requirements Overview](#1-requirements-overview)
+    * [1.1 Functional requirements](#11-functional-requirements)
+    * [1.2 CLI requirements](#12-cli-requirements)
+    * [1.3 Scalability and Default Values](#13-scalability-and-default-values)
+    * [1.4 Warm Restart requirements ](#14-warm-restart-requirements)
+  * [2 Architecture Design](#2-architecture-design)
+    * [2.1 High-Level Architecture](#21-high-level-architecture)
+    * [2.2 OpenBMP](#22-openbmp)
+    * [2.3 BGP/FRR Update](#23-bgp/frr-update) 
+    * [2.4 Database Design](#24-database-design)
+    * [2.5 BMP Agent](#25-bmp-agent)
+    * [2.6 GNMI support](#26-gnmi-support)
+  * [3 CLI](#3-cli)
+  * [4 Resource usage and Test plan](#4-resource-usage-and-test-plan)
+
+###### Revision
+
+| Rev |     Date    |       Author       | Change Description                |
+|:---:|:-----------:|:------------------:|-----------------------------------|
+| 0.1|  02/20/2024  |     Feng Pan       | Initial version                   |
+
+
+# About this Manual
+This document will propose to bring up BMP container on SONiC, which is forked from openbmp project with some code changes, by that we could improve the SONiC debuggability and BGP service monitoring efficiency.
+
+# Definitions/Abbreviation
+
+| **Term**                 | **Meaning**                         | **Link**                         |
+|--------------------------|-------------------------------------|--------------------------|
+| BMP                      | BGP Monitor Protocol                |https://datatracker.ietf.org/doc/html/rfc7854 |
+| OpenBMP                  | Open Source BGP Monitoring Protocol (OpenBMP) Collection Framework    |https://github.com/SNAS/openbmp|
+| BGP/FRR                  | SONiC BGP container and bgpd daemon which will collborate with BMP here   |https://github.com/sonic-net/sonic-frr|
+| GNMI                     | gRPC Network Management Interface   |https://github.com/openconfig/gnmi|
+
+
+# 1 Requirements Overview
+
+## 1.1 Functional requirements
+
+Now SONiC supports standard BGP session established with BGP neighbors, but there's only limited BGP info readable in database, which makes some cases difficult to debug, we need to dump more BGP data, meanwhile we're also expecting some live debug option which could dump BGP related info on demand and improve the debuggability.
```

Comment:
VTYSH show commands provide extensive dumps today for the debuggability, trying to understand the new information that we support part of openBMP.

On-change subscription for BGP data is clearly an use-case, I would like to see more filtering options (e.g RIB-in or our per neighbor) in the HLD so that we can limit the number entries going into the DB.

Reply:
Are you asking smaller granularity control like per neighbor? instead of per table?.

Reply:
I'm asking per neighbor level filtering at least and if possible, per table level filtering is good as well.

Reply:
Sure, we will support table level filtering first, later could append neighbor level filtering if specific requirement asks.